pulp_puppet_plugins/pulp_puppet/plugins/migrations/0005_puppet_module_name_change.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+from mongoengine import NotUniqueError
+
+from pulp.server.controllers import repository as repo_controller
+from pulp.server.db import model
+
+from pulp_puppet.plugins.db.models import Module
+
+_log = logging.getLogger('pulp')
+
+
+def migrate(*args, **kwargs):
+    """
+    For each puppet module check and if needed update module name format.
+    """
+    modules = Module.objects(__raw__={'name': {'$regex': '/|-'}})
+    repos_to_rebuild = set()
+    for puppet_unit in modules:
+        try:
+            author, name = puppet_unit['name'].split("-", 1)
+        except ValueError:
+            # This is the forge format, but Puppet still allows it
+            author, name = puppet_unit['name'].split("/", 1)
+        try:
+            puppet_unit.update(set__name=name)
+        except NotUniqueError:
+            # find all repos that have this unit
+            repos_with_unit = model.RepositoryContentUnit.objects(unit_id=puppet_unit.id)
+            repos_to_rebuild.update(repos_with_unit)
+            # find unit with correct name
+            correct_unit = Module.objects(__raw__={'name': name})
+            for repo in repos_with_unit:
+                # unassociate wrong unit
+                repo_controller.disassociate_units(repo, [puppet_unit])
+                # associate correct unit to the list of the repos
+                repo_controller.associate_single_unit(repo, correct_unit[0])
+
+    repo_list = []
+    for repo in repos_to_rebuild:
+        repo_obj = model.Repository.objects.get_repo_or_missing_resource(repo.repo_id)
+        repo_controller.rebuild_content_unit_counts(repo_obj)
+        repo_list.append(repo.repo_id)
+
+    repos_to_republish = model.Distributor.objects(repo_id__in=repo_list, last_publish__ne=None)
+    # redirect output to file
+    f = open('/tmp/repos_to_republish.txt', 'w')
+    f.write(str([repo.repo_id for repo in repos_to_republish]))
+    f.close()
+    _log.info("You need to re-publish list of repos found in %s" % f.name)
```

Comment:
Also a little more explanation to the user would be good. They don't have to re-publish, but they can choose to. They would want to do so to resolve issues when puppet modules have trouble being installed due to the name not being correct which this migration fixed.